test/test_supported_http_methods.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+require_relative "helper"
+
+require "puma/server"
+
+class TestHandler
+  attr_reader :ran_test
+
+  def call(env)
+    @ran_test = true
+
+    [200, {"Content-Type" => "text/plain"}, ["hello!"]]
+  end
+end
+
+class WebServerTest < Minitest::Test
```

Comment:
I think you should look over the existing tests, looks like you have invented the wheel again for a few test helpers things in this file

Reply:
I might need your guidance here. 

I wanted to have a test case where I could pass the `supported_http_methods` option when instantiating the server, but in the other test files, the instantiation was done in the `setup` method, which is implicitly called and taks no parameter.

This is why I introduced the `run_server` method, which takes an `options:` parameter.

Is there a more elegant way to test the behavior for different values of the `supported_http_methods` option?

Thank you very much.

Reply:
OK, I spent a bit more time and simplified the helpers quite a bit here!

Let me know if there is yet a more simple way to do things. :pray:

Reply:
Looks pretty clean to me now but I suspect it could be refactored to share even more with existing helpers, I haven't looked in detail though, so let' save that for some future contributor (if even possible) :)